services/src/main/java/edu/unc/lib/dl/cdr/services/rest/modify/ExportCsvController.java

```diff
@@ -173,14 +175,47 @@
         return null;
     }
 
+    private String computePatronPermissions(List<String> roles) {
+        String permission = "Restricted";
+
+        if (roles == null) {
+            return permission;
+        }
+
+        Map<String, String> roleList = new HashMap<>();
+        for (String role : roles) {
+            String[] principalRole = role.split("\\|");
+            roleList.put(principalRole[1], principalRole[0]);
+        }
+
+        String everyoneRole = roleList.get("everyone");
+        String authenticatedRole = roleList.get("authenticated");
+
+        if (canViewOriginals(everyoneRole)) {
+            permission = "Public";
+        } else if (canViewOriginals(authenticatedRole) && hasNoAccess(everyoneRole)) {
+            permission = "Authenticated";
+        } else if (hasNoAccess(everyoneRole) && hasNoAccess(authenticatedRole)) {
+            permission = "Staff-only";
+        }
+
+        return permission;
+    }
+
+    private boolean canViewOriginals(String role) {
+        return role != null && role.equals("canViewOriginals");
```

Comment:
use the constant `edu.unc.lib.dl.acl.util.UserRole.canViewOriginals.getPredicate()`